File: MovieServer/MovieServer.API/MovieServer.Infrastructure/src/Shared/Repositories/CartRepo.ts
import { Cart } from "@Domain/Entities";
import { ICartRepo } from "@Domain/Repositories";
import { CartId } from "@Domain/ValueObjects";
import { ICacheService } from "@Infrastructure/Shared/Services";
import { CacheService, Injectable, Mapper } from "@Shared/IoC";
import { AutoMapper } from "@Shared/AutoMapper";
import { CartReadDto } from "@Application/DTO";

@Injectable
export default class CartRepo implements ICartRepo {

    private readonly _cacheService: ICacheService;
    private readonly _mapper: AutoMapper;
    constructor(
        @CacheService cacheService: ICacheService,
        @Mapper mapper: AutoMapper
    ) {
        this._cacheService = cacheService;
        this._mapper = mapper;
    }
    async Get(id: CartId): Promise<Cart | null> {
        const data = await this._cacheService.Get<CartReadDto>(id.Guid);
        if (data === null) return null;
        return this._mapper.map(data, CartReadDto, Cart);
    }
    async Create(cart: Cart): Promise<any> {
        //default time to leave
        const data = this._mapper.map(cart, Cart, CartReadDto);
        await this._cacheService.Set(cart.Id.Guid, data, Cart.TIME_TO_LEAVE);
    }
    async Update(cart: Cart): Promise<any> {
        const data = this._mapper.map(cart, Cart, CartReadDto);
        await this._cacheService.Set(cart.Id.Guid, data, Cart.TIME_TO_LEAVE);
    }
    async Remove(cart: Cart): Promise<any> {
        await this._cacheService.Remove(cart.Id.Guid);
    }

}

File: OrderServer/OrderServer.API/src/OrderServer.Infrastructure/Shared/Repositories/CartRepo.cs
using OrderServer.Domain.Entities;
using OrderServer.Domain.Repositories;
using OrderServer.Domain.ValueObjects;
using OrderServer.Shared.Cache;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OrderServer.Infrastructure.Shared.Repositories
{
    public class CartRepo : ICartRepo
    {
        private readonly ICacheService _cacheService;
        public CartRepo(
            ICacheService cacheService)
        {
            _cacheService = cacheService;
        }
        public Task DeleteAsync(Cart cart)
        {
            var key = cart.Id.Guid.ToString();
            return _cacheService.DeleteAsync(key);
        }

        public Task<Cart?> GetAsync(CartId id)
        {
            var key = id.Guid.ToString();
            return _cacheService.GetAsync<Cart>(key);
        }
    }
}
